Lib/dataclasses.py

```diff
@@ -1445,3 +1447,15 @@ class C:
     # changes that aren't fields, this will correctly raise a
     # TypeError.
     return obj.__class__(**changes)
+
+def add_field_docs(cls, doc):
+    if not cls.__dataclass_fields__:
+        return doc
+    if doc:
+        doc += '\n'
+    for name, f in cls.__dataclass_fields__.items():
+        dflt = f' [{f.default}]' if f.default is not MISSING else ' '
+        fdoc = f' -- {f.doc}' if f.doc else ''
+        doc += f'\n{name}: {f.type}{dflt}{fdoc}\n'
+    doc = doc[:-1]
+    return doc
```

Comment:
I'd suggest using `fields(cls)` instead of relying on `__dataclass_fields__`:
```suggestion
def add_field_docs(cls, doc):
    if doc:
        doc += '\n'
    for f in fields(cls):
        dflt = f' [{f.default}]' if f.default is not MISSING else ' '
        fdoc = f' -- {f.doc}' if f.doc else ''
        doc += f'\n{f.name}: {f.type}{dflt}{fdoc}\n'
    doc = doc[:-1]
    return doc
```
(Not sure about string concatenation with `+=`, might be better to use a `_field_doc(f)` helper function and `'\n'.join(...)`.)